packages/react/src/publication/adapters/ReactionPresenter.ts

```diff
@@ -39,46 +50,37 @@ export class ReactionPresenter
   }
 
   private async addReaction(request: ReactionRequest): Promise<void> {
-    switch (request.reactionType) {
-      case ReactionType.UPVOTE:
-        {
-          this.publicationCacheManager.update(request.publicationId, (current) => ({
-            ...current,
-            stats: {
-              ...current.stats,
-              totalUpvotes: current.stats.totalUpvotes + 1,
-            },
-            reaction: ReactionTypes.Upvote,
-          }));
-        }
-        break;
-
-      case ReactionType.DOWNVOTE:
-        throw new Error('Downvotes support not implemented');
-      default:
-        throw new Error('Unknown reaction type');
-    }
+    this.publicationCacheManager.update(request.publicationId, (current) => {
+      const removedStatKey = current.reaction
```

Comment:
what is `removedStatKey` for?

Reply:
If we add a downvote to a publication that already has an upvote, we have to remove 1 from the stat value that the active profile previously had -- `removedStatKey` is that key in `PublicationStats` (`totalUpvotes` | `totalDownvotes`)